datm/datm_datamode_simple_mod.F90

```diff
@@ -0,0 +1,385 @@
+    real(R8) :: rday               ! elapsed day
+    real(R8) :: cosFactor          ! cosine factor
+    real(R8) :: factor             ! generic/temporary correction factor
+    real(R8) :: tMin               ! minimum temperature
+    real(R8) :: uprime,vprime
+    character(len=*), parameter :: subname='(datm_datamode_simple): '
+    !-------------------------------------------------------------------------------
+
+    rc = ESMF_SUCCESS
+
+    lsize = size(Sa_z)
+
+    call shr_cal_date2julian(target_ymd, target_tod, rday, model_calendar)
+    rday = mod((rday - 1.0_R8),365.0_R8)
+    cosfactor = cos((2.0_R8*SHR_CONST_PI*rday)/365 - phs_c0)
+
+    do n = 1,lsize
+      Sa_z(n) = 10.0_R8
+
+      !--- (i) Set forcing fields to constant values read from the namelist file ---
+      Sa_dens(n) = dn10
+      Sa_pslv(n) = slp
+      Sa_pbot(n) = Sa_pslv(n)
+      Sa_shum(n) = q
+      Sa_tbot(n) = t
+      Sa_ptem(n) = Sa_tbot(n)
+      Sa_u(n) = u
+      Sa_v(n) = v
+
+      !--- (ii) Set precipitation (currently all zeros) ---
+
+      Faxa_rainc(n) = 0.0_R8               ! default zero
+      Faxa_snowc(n) = 0.0_R8
+      if (Sa_tbot(n) < tKFrz ) then        ! assign precip to rain/snow components
+         Faxa_rainl(n) = 0.0_R8
+         Faxa_snowl(n) = 0.0_R8 ! todo
+      else
+         Faxa_rainl(n) = 0.0_R8 ! todo
+         Faxa_snowl(n) = 0.0_R8
+      endif
+
+      !--- (iii) RADIATION DATA ---
+
+      !--- fabricate required swdn components from net swdn ---
+      Faxa_swvdr(n) = strm_swdn(n)*(0.28_R8)
+      Faxa_swndr(n) = strm_swdn(n)*(0.31_R8)
+      Faxa_swvdf(n) = strm_swdn(n)*(0.24_R8)
+      Faxa_swndf(n) = strm_swdn(n)*(0.17_R8)
+      !--- compute net short-wave based on LY08 latitudinally-varying albedo ---
+      avg_alb = ( 0.069 - 0.011*cos(2.0_R8*yc(n)*degtorad ) )
+      Faxa_swnet(n) = strm_swdn(n)*(1.0_R8 - avg_alb)
+      !--- corrections to GISS sswdn for heat budget balancing ---
+      factor = 1.0_R8
+      if      ( -60.0_R8 < yc(n) .and. yc(n) < -50.0_R8 ) then
+         factor = 1.0_R8 - (yc(n) + 60.0_R8)*(0.05_R8/10.0_R8)
+      else if ( -50.0_R8 < yc(n) .and. yc(n) <  30.0_R8 ) then
+         factor = 0.95_R8
+      else if (  30.0_R8 < yc(n) .and. yc(n) <  40._R8 ) then
+         factor = 1.0_R8 - (40.0_R8 - yc(n))*(0.05_R8/10.0_R8)
+      endif
+      Faxa_swnet(n) = Faxa_swnet(n)*factor
+      Faxa_swvdr(n) = Faxa_swvdr(n)*factor
+      Faxa_swndr(n) = Faxa_swndr(n)*factor
+      Faxa_swvdf(n) = Faxa_swvdf(n)*factor
+      Faxa_swndf(n) = Faxa_swndf(n)*factor
+      !--- correction to GISS lwdn in Arctic ---
+      if ( yc(n) > 60._R8 ) then
+         factor = MIN(1.0_R8, 0.1_R8*(yc(n)-60.0_R8) )
+         Faxa_lwdn(n) = Faxa_lwdn(n) + factor * dLWarc
+      endif
+
+    enddo   ! lsize
+
+    if (associated(Faxa_ndep)) then
+       ! convert ndep flux to units of kgN/m2/s (input is in gN/m2/s)
+       Faxa_ndep(:,:) = Faxa_ndep(:,:) / 1000._r8
+    end if
+
```

Comment:
Can we name all of the constants here and use parameters to define them?  For example it looks like line 323 is a range of latitude values.  I think that it would be clearer to have
```
real(r8), parameter :: maxlat = -60.0_r8
real(r8), parameter :: minlat = -50.0_r8

if (maxlat < yc(n) .and. yc(n) < minlat) then

```